File: acmp/arch/4.Вычислительная-геометрия/0822.Площадь-треугольника/main.cpp
#include <cmath>
#include <iomanip>
#include <iostream>
#include <fstream>


using namespace std;





char const *INPUT_FILE_NAME = "INPUT.TXT";
char const *OUTPUT_FILE_NAME = "OUTPUT.TXT";





int main( int argc, char *argv[] )
{
	double x1, y1, x2, y2, x3, y3;
	ifstream(INPUT_FILE_NAME) >> 
		x1 >> y1 >>
		x2 >> y2 >>
		x3 >> y3;

	double
		len1 = hypot( x2-x1, y2-y1 ),
		len2 = hypot( x3-x2, y3-y2 ),
		len3 = hypot( x1-x3, y1-y3 );

	// heron
	double p = ( len1 + len2 + len3 ) / 2.0f;
	double s = sqrt( p * ( p-len1 ) * ( p-len2 ) * ( p-len3 ) );

	

	ofstream( OUTPUT_FILE_NAME ) <<
		fixed << setprecision(1) << s;

	return 0;
}





// end

File: acmp/arch/4.Вычислительная-геометрия/0822.Площадь-треугольника/main.py

strlist = open("INPUT.TXT", 'r').read().split();

x1 = float(strlist[0]);
y1 = float(strlist[1]);
x2 = float(strlist[2]);
y2 = float(strlist[3]);
x3 = float(strlist[4]);
y3 = float(strlist[5]);


len1 = ( (x2-x1)**2 + (y2-y1)**2 )**0.5;
len2 = ( (x3-x2)**2 + (y3-y2)**2 )**0.5;
len3 = ( (x1-x3)**2 + (y1-y3)**2 )**0.5;

p = ( len1 + len2 + len3 ) / 2.0;
s = (p*(p-len1)*(p-len2)*(p-len3))**0.5;

open("OUTPUT.TXT", 'w').write( str(s) );


